scraper.rb

```diff
@@ -58,7 +58,7 @@ def wikidata_id(url)
     constituency:                r[:constituencylabel],
     constituency_id:             r[:constituencylabel][/(NA-[0-9]+)/, 1] || wikidata_id(r[:constituency]),
     party:                       r[:partyshortname].to_s.empty? ? r[:partylabel] : r[:partyshortname],
-    party_id:                    r[:partyshortname].to_s.empty? ? wikidata_id(r[:party]) : r[:partyshortname].downcase ,
+    party_id:                    r[:partyshortname].to_s.empty? ? wikidata_id(r[:party]) : r[:partyshortname].downcase.gsub(/[^a-z]/, '') ,
```

Comment:
This certainly _works_, but if you're keeping as a `gsub` with a regex, `\W+` is probably preferable to an explicit range like this. However, for this sort of character-by-character replacement, `tr` might be more suitable:`.tr('^a-z', '')`

NB: probably best to clean the spurious space before the end there too (even though you didn't introduce it in this change)